src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -11893,43 +11893,308 @@ static bool Dummy(object y, object z)
+        AssertFalse(Box((uint)(4294967295U)) is (long)(2147483648L));
+        AssertFalse(Box((long)(-9223372036854775808L)) is (ushort)(0));
+        AssertFalse(Box((long)(-9223372036854775808L)) is (uint)(2147483648U));
+        AssertFalse(Box((long)(-9223372036854775808L)) is (long)(1L));
+        AssertFalse(Box((long)(-9223372036854775808L)) is (long)(2147483648L));
+        AssertFalse(Box((long)(0L)) is (int)(1));
+        AssertFalse(Box((long)(0L)) is (ulong)(2147483647UL));
+        AssertFalse(Box((long)(2147483647L)) is (uint)(1U));
+        AssertFalse(Box((long)(2147483648L)) is (sbyte)(127));
+        AssertFalse(Box((long)(2147483648L)) is (uint)(0U));
+        AssertFalse(Box((long)(9223372036854775807L)) is (byte)(0));
+        AssertFalse(Box((long)(9223372036854775807L)) is (long)(-9223372036854775808L));
+        AssertFalse(Box((ulong)(0UL)) is (ushort)(1));
+        AssertFalse(Box((ulong)(0UL)) is (uint)(1U));
+        AssertFalse(Box((ulong)(0UL)) is (uint)(2147483647U));
+        AssertFalse(Box((ulong)(0UL)) is (long)(1L));
+        AssertFalse(Box((ulong)(0UL)) is (ulong)(2147483648UL));
+        AssertFalse(Box((ulong)(1UL)) is (ushort)(65535));
+        AssertFalse(Box((ulong)(1UL)) is (int)(2147483647));
+        AssertFalse(Box((ulong)(1UL)) is (long)(-9223372036854775808L));
+        AssertFalse(Box((ulong)(2147483647UL)) is (sbyte)(0));
+        AssertFalse(Box((ulong)(2147483647UL)) is (long)(-9223372036854775808L));
+        AssertFalse(Box((ulong)(2147483648UL)) is (byte)(0));
+        AssertFalse(Box((ulong)(2147483648UL)) is (short)(-32768));
+        AssertFalse(Box((ulong)(2147483648UL)) is (ushort)(1));
+        AssertFalse(Box((ulong)(2147483648UL)) is (int)(1));
+        AssertFalse(Box((ulong)(2147483648UL)) is (uint)(0U));
+        AssertFalse(Box((ulong)(18446744073709551615UL)) is (byte)(0));
+        AssertFalse(Box((ulong)(18446744073709551615UL)) is (sbyte)(1));
+        AssertFalse(Box((ulong)(18446744073709551615UL)) is (sbyte)(127));
+        AssertFalse(Box((ulong)(18446744073709551615UL)) is (short)(1));
+        AssertFalse(Box((ulong)(18446744073709551615UL)) is (ushort)(1));
+        AssertFalse(Box((ulong)(18446744073709551615UL)) is (ulong)(0UL));
+        System.Console.WriteLine(""done"");
+    }
+    private static object Box(object o) => o;
+    private static void AssertTrue(bool shouldBeTrue)
+    {
+        if (!shouldBeTrue) throw null;
+    }
+    private static void AssertFalse(bool shouldBeFalse)
+    {
+        if (shouldBeFalse) throw null;
     }
 }
 ";
             var compilation = CreateCompilationWithMscorlib(source, options: TestOptions.ReleaseExe, parseOptions: patternParseOptions);
             compilation.VerifyDiagnostics();
-            var verifier = CompileAndVerify(compilation, expectedOutput: @"
-True
-True
-False
-False
-False");
+            var verifier = CompileAndVerify(compilation, expectedOutput: @"done");
         }
+
```

Comment:
You may want to leave a comment on how these were generated in the source, in case someone wants to do something similar again.